### ISSUE TITLE
Webhooks fail when multiple inserts are done

### ISSUE THREAD

**Issue by devdras** [supabase/pg_net#86]
# Bug report

## Describe the bug

I have a webhook configured to watch a table for inserts. On an insert a POST request should be fired to an API hosted on Vercel. 
This works when one insert operation is carried out. If multiple inserts are carried out: .insert([{message: 1}, {message: 2}, {message: 3}]) then the webhooks aren't fired at all. If instead we carry out multiple inserts by calling the Supabase API multiple times then the webhook is fired for only some of the inserts. 

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

1. Configure a webhook on a table to watch for insert operations and call an external API
2. Use the Supabase JS library to perform multiple inserts like: `let { data, error } = await supabase
              .from("example_table")
              .insert([{message: 1}, {message: 2}, {message: 3}, {message: 4}, {message: 5}, {message: 6}, etc ])
              .select("message");`

4. Check logs for external API to confirm it didn't receive any requests
5. See error

Something to note is that I was inserting 32 things at once.

## Expected behavior

For each insert, I would expect the external API to be called. It would also help if there was a log in Supabase that showed what happens when the webhook is run.

## System information

- OS: N/A
- Browser (if applies): N/A
- Version of supabase-js: 2.0.0-rc.8
- Version of Node.js: v16.17.0 on desktop, 18.x on Vercel (per their docs)

**Comment by steve-chavez** [comment]
We've tested pg_net sending 1K inserts per second, this to a local nginx server though.

@devdras @danrasmuson Which webhook service are you using? Likely it's rate limiting the requests.

**Comment by danrasmuson** [comment]
> We've tested pg_net sending 1K inserts per second, this to a local nginx server though.
> 
> 
> 
> @devdras @danrasmuson Which webhook service are you using? Likely it's rate limiting the requests.

Im using vercel with serverless functions

**Comment by matt-aitken** [comment]
We're experiencing the same problem. When inserting records in sequence all the webhooks are received. But if we insert 30 rows in parallel none of the webhooks send and the `net/_http_response` table is full of `Timeout was reached` errors.

**Comment by steve-chavez** [comment]
@matt-aitken Are you using vercel serverless functions too?

**Comment by lohrmann** [comment]
@steve-chavez We're able to reliably reproduce it based on the number of new entries that get bulk added to the table—have been in touch with your support team about this as well, but unable to resolve it. 
- The webhook trigger is calling an AWS Lambda function
- The lambda function does not record any invocations if the number of records inserted is larger than >150
  - This is an **all or nothing behaviour**. If we insert fewer records, all invocations happen, if we insert more records none of the invocations happen
  - It's **not** the case that only some of the invocations don't come through
  - It's **not** the case that the invocation reaches the Lambda function and the webhook times out before the function responds
- In supabase, the _http_response table only logs "Timeout was reached", nothing else
- We have tested this with different webhooks — including [webhook.site](http://webhook.site/). No invocations of the webhook are received

Happy to prepare a screen-recording or do a walk through if helpful.

**Comment by steve-chavez** [comment]
> The lambda function does not record any invocations if the number of records inserted is larger than >150

@lohrmann That might be related to lambdas rate limiting https://docs.aws.amazon.com/lambda/latest/dg/gettingstarted-limits.html#api-requests

> We're able to reliably reproduce it based on the number of new entries that get bulk added to the table

By default there are 200 requests per iteration https://github.com/supabase/pg_net/blob/master/src/worker.c#L479-L483

I'll make this parameter user configurable.

**Comment by steve-chavez** [comment]
Curl has:

```bash
curl --rate 2/s -O https://example.com/[1-100].jpg
```

https://everything.curl.dev/usingcurl/transfers/request-rate#examples

I think that should be the final solution here.

---

Note: the above is not available on `libcurl`.

**Comment by matt-aitken** [comment]
Sorry for the slow response on this. No, we're receiving webhooks to AWS EC2 (a long-running server).

There's nothing in our server logs where we receive requests from Supabase (when inserting many rows) and the `net/_http_response` table is full of `Timeout was reached` errors.

We (Trigger.dev) make it easy for our customers to subscribe to these Supabase database webhooks and quite a few people are hitting this problem now.

**Comment by ahanusek** [comment]
> We're experiencing the same problem. When inserting records in sequence all the webhooks are received. But if we insert 30 rows in parallel none of the webhooks send and the `net/_http_response` table is full of `Timeout was reached` errors.

I have the same issue. I'm using Google Cloud Functions v2 (Cloud Run). Table is full of `Timeout was reached`, but in Google Cloud Monitoring don't have any logs.

**Comment by reedshea** [comment]
I'm running into this issue as well. I spun up a clean environment to reproduce, and did some testing. Supabase project `qutskxfebtlgwyzsonvd`. The setup is:

* A `users` table
* A `queue` table
* A cron job that inserts one item into the `queue` for each user
* A database webhook that sends the `queue` insert to an HTTPS endpoint, an AWS API Gateway 
* AWS API Gateway configured for [asynchronous invocation](https://docs.aws.amazon.com/lambda/latest/dg/invocation-async.html), which always returns a `200` response (typically in <80ms)
* AWS Lambda function that is just `sleep(5)`

I ran a handful of tests ([spreadsheet here](https://docs.google.com/spreadsheets/d/1d-b3jawq_loBELlkNs5KBdiSlbmepSedqGfMSyeZ04U/edit?usp=sharing), screenshot below). It looks like a pretty linear relationship between the number of requests (in my case, the number of `users` for which the cron job inserts items into the `queue`) and the webhook timeout configuration. By my math, the webhook needs about 33.33ms per item to process. If only 20ms per item is available within the webhook timeout configuration, most or all of the events fail to be sent to the API Gateway.

Unlike the note from @lohrmann [above](https://github.com/supabase/pg_net/issues/86#issuecomment-1760574172), I did see some partial successes.

<img width="593" alt="Screenshot 2024-01-06 at 6 11 53 PM" src="https://github.com/supabase/pg_net/assets/4605293/3c950e52-dfc8-48de-be48-ec1b0273feb1">

**Comment by steve-chavez** [comment]
@reedshea Thanks for the detailed information.

@TheOtherBrian1 Shared a reproducible example: https://gist.github.com/TheOtherBrian1/e479426f0070d5f0879483cf59f85dfb

The setup is basically:

```sql
CREATE TABLE webhook_test (
    id SERIAL,
    sample text
);

CREATE OR REPLACE FUNCTION debugging_example()
RETURNS TRIGGER
AS $$
DECLARE
    -- Store payload data
    row_data_var JSONB;
    row_id INT;
BEGIN
    -- Retrieve row data and convert to JSON
    SELECT TO_JSONB(NEW.*) INTO row_data_var;
    -- Initiate HTTP POST request to URL
    SELECT
        net.http_post(
            url := 'https://postman-echo.com/post',
            -- Use row data as payload
            body := row_data_var
        ) INTO row_id;

    RAISE LOG 
        'LOGGING WEBHOOK |
        The Webhook was called after %s |
        ID = %s |
        Logging an entire row as JSON (%)', (SELECT NOW()), row_id, row_data_var;
    -- Return the row data to inspect
    RETURN NEW;
-- Handle exceptions here if needed
EXCEPTION
    WHEN others THEN
        RAISE EXCEPTION 'LOGGING WEBHOOK FAILED: An error occurred: %', SQLERRM;
END;
$$ LANGUAGE plpgsql;


CREATE TRIGGER webhook
    AFTER INSERT ON webhook_test
    FOR EACH ROW
    EXECUTE FUNCTION debugging_example();

CREATE OR REPLACE FUNCTION insert_test_data(num_rows INT)
RETURNS VOID
AS $$
DECLARE
    i INT;
BEGIN
    FOR i IN 1..num_rows LOOP
        INSERT INTO webhook_test(sample) VALUES (i || 'Call');
    END LOOP;
END;
$$ LANGUAGE plpgsql;
```

Then execute a request:
```sql
INSERT INTO webhook_test (sample)
VALUES ('test value');
```

Then a batch of requests:
```sql
SELECT insert_test_data(100);
```

Locally, all requests succeed fast:

```sql
select count(*) from net._http_response where error_msg is null;
-[ RECORD 1 ]
count | 101

select count(*) from net._http_response where error_msg is not null;
-[ RECORD 1 ]
count | 0
```

But, for some reason, on Supabase cloud only the single request succeeds and the batched ones fail:

```sql
select count(*) from net._http_response where error_msg is null;
```
| count |
| ----- |
| 1     |

```sql
select count(*) from net._http_response where error_msg is not null;
```
| count |
| ----- |
| 100   |

**Comment by steve-chavez** [comment]
By increasing the default timeout (`2 seconds`) to `4 seconds`, all the [above](https://github.com/supabase/pg_net/issues/86#issuecomment-1886040988) requests succeed on Supabase cloud:

```sql
        net.http_post(
            url := 'https://postman-echo.com/post',
            -- Use row data as payload
            body := row_data_var,
            timeout_milliseconds := 4000
        ) INTO row_id;
```

```sql
SELECT insert_test_data(100);

select count(*) from net._http_response where error_msg is null;
```

| count |
| ----- |
| 100   |

Will increase the default to 5 seconds.

**Comment by vickkhera** [comment]
> Will increase the default to 5 seconds.

Is that setting the timeout for the whole batch? It seems based on @reedshea's graph, the whole batch needs enough time per call. Or is he setting some other timeout?

**Comment by TheOtherBrian1** [comment]
@vickkhera I wasn't actually aware of the cause of the bug - I was only aware of how to reproduce it - but you seem to have noticed that the timeout is caused when a batch (default 200) does not complete in the set timeout period. A later batch would actually succeed if its queue were smaller or the timeout were higher. 

I assumed that the timeout setting was for an individual cURL requests and not the entire batch. @steve-chavez, is this the intended behavior?

**Comment by steve-chavez** [comment]
The timeouts are per-request. Will add a test to confirm soon.

**Comment by vickkhera** [comment]
> The timeouts are per-request. Will add a test to confirm soon.

There's no way my requests were taking 2s each. The whole batch timed out in ~2-3 seconds.

**Comment by andrueandersoncs** [comment]
We're running into this same issue. "net._http_response" table shows the "Timeout was reached" error yet the "timed_out" column remains null. This is happening with batches of 8 or less, and it seemed like after the initial failure subsequent inserts into the table with the trigger failed even with a batch of 1.

**Comment by ham-evans** [comment]
I'm having this problem as well. Can reproduce it based on the number of new entries that get bulk added to the table. Around 10 work, more than that get a significant % of errors

- The webhook trigger is calling an AWS Lambda function (via Gateway).
- The lambda function does not record any invocations if the number of records inserted 
- It's not the case that only some of the invocations don't come through
- It's not the case that the invocation reaches the Lambda function and the webhook times out before the function responds
- Other times it doesn't time out and just returns a 500 with "Internal Server Error"
- In supabase, the _http_response table logs everything

All using pgnet and supabase webhooks. Would love if theres some solution? Using the lambda to then load SQS. Thinking of skipping this step now...

**Comment by waylon999** [comment]
> I'm having this problem as well. Can reproduce it based on the number of new entries that get bulk added to the table. Around 10 work, more than that get a significant % of errors
> 
> * The webhook trigger is calling an AWS Lambda function (via Gateway).
> * The lambda function does not record any invocations if the number of records inserted
> * It's not the case that only some of the invocations don't come through
> * It's not the case that the invocation reaches the Lambda function and the webhook times out before the function responds
> * Other times it doesn't time out and just returns a 500 with "Internal Server Error"
> * In supabase, the _http_response table logs everything
> 
> All using pgnet and supabase webhooks. Would love if theres some solution? Using the lambda to then load SQS. Thinking of skipping this step now...

I am doing something similar, except API Gateway populates SQS which then calls a lambda function. Worked fine on small data sets but when I had to make over 100 calls I saw similar behavior as to what you (and others) describe. I set the timeout to 5 seconds and it is working now for me, although that feels very fragile. Who knows if it will work if I have a big burst of calls I need to make. The UI says 5000 ms is the max timeout, but I believe you can set it higher than that if you just create the function via SQL editor, though I'm not sure if that has any side effects.

**Comment by ham-evans** [comment]
> > I'm having this problem as well. Can reproduce it based on the number of new entries that get bulk added to the table. Around 10 work, more than that get a significant % of errors
> > 
> > * The webhook trigger is calling an AWS Lambda function (via Gateway).
> > * The lambda function does not record any invocations if the number of records inserted
> > * It's not the case that only some of the invocations don't come through
> > * It's not the case that the invocation reaches the Lambda function and the webhook times out before the function responds
> > * Other times it doesn't time out and just returns a 500 with "Internal Server Error"
> > * In supabase, the _http_response table logs everything
> > 
> > All using pgnet and supabase webhooks. Would love if theres some solution? Using the lambda to then load SQS. Thinking of skipping this step now...
> 
> I am doing something similar, except API Gateway populates SQS which then calls a lambda function. Worked fine on small data sets but when I had to make over 100 calls I saw similar behavior as to what you (and others) describe. I set the timeout to 5 seconds and it is working now for me, although that feels very fragile. Who knows if it will work if I have a big burst of calls I need to make. The UI says 5000 ms is the max timeout, but I believe you can set it higher than that if you just create the function via SQL editor, though I'm not sure if that has any side effects.

yea i have it set to 5000ms and still having the problem.  Agree it seemed fragile regardless.  would love to know if doing via SQL Editor solves? 

Otherwise gonna basically skip the webhook altogether as can't risk it in production

**Comment by TheOtherBrian1** [comment]
@waylon999 You can set the timeout to greater than 5000ms. That's a limitation imposed by the UI. You can write the webhooks directly in pg_net or use the [supabase_functions.http_request](https://supabase.com/docs/guides/database/webhooks) helper function.

**Comment by ahanusek** [comment]
This bug has been open for a year, and it's really frustrating that issues like this are not properly addressed by the Supabase team. They are adding other advanced features like branching, but the core remains unstable, and I don't see a significant effort to fix that. I migrated a relatively small project from Firestore/Firebase to Supabase because I believed in their marketing, and now I really regret this decision. Currently, I'm paying more for significantly worse quality:

- Webhooks are unstable. In my previous architecture, I heavily relied on Firestore triggers to perform some operations in the background. Now, they can randomly fail without any option to retry, which is a HUGE problem when something needs to happen in the background of our app.

- Realtime functionality is [useless](https://github.com/orgs/supabase/discussions/5641); it only works fine when you have an active tab in the browser. Then, you need to figure out the whole mechanism for refetching data and reconnecting after you go from inactive. I don't even want to mention how painful this is on mobile (React Native). With Firebase, I didn't even have to think about it; it just worked out of the box.

- Pooler / pg_bouncer - Random failures when connecting to the DB from cloud functions / AWS Lambda (Connection terminated unexpectedly).

- Upgrading your [db](https://github.com/supabase/pg_net/pull/129)? Okay, our app will be unavailable for minutes or hours, and then you need to figure out why something doesn't work.

- [Storage](https://supabase.com/docs/reference/javascript/storage-from-upload?example=upload-file-using-arraybuffer-from-base64-file-data) - You can upload from RN, but first, you need to convert the entire file into base64 and then use ArrayBuffer.

But yes, Supabase is open-sourced. I guess I'll tell my clients - our app doesn't work properly, but at least you can read the source code.

**Comment by ham-evans** [comment]
> @waylon999 You can set the timeout to greater than 5000ms. That's a limitation imposed by the UI. You can write the webhooks directly in pg_net or use the [supabase_functions.http_request](https://supabase.com/docs/guides/database/webhooks) helper function.

does this solve the problem though? Seems fragile regardless of timeout (esp if unclear if timeout will be 5 seconds or 30)

**Comment by joaopegoraro** [comment]
Any new updates on this?

**Comment by robertbodley** [comment]
Any new updates on this?

**Comment by kevinlooESO** [comment]
pg_net has been super inconsistent. also tends to hang up sporadically. tried to build an event-driven app but decide to give up & rely on cron job instead

**Comment by matiastucci** [comment]
Is there any ETA on this? It's blocking me and I would need to do some workaround. Thanks!

**Comment by ExplosiveJam** [comment]
Experiencing same issue. I increased timeout to 30s and all requests ended up with either "SSL connect error" or "Error in the HTTP2 framing layer".

**Comment by uncvrd** [comment]
I'm unfortunately dealing with this as well. For me it's when there are 100 concurrent events (like bulk deleting 100 rows). I'm curious what alternatives you all are using for listening to postgres events and performing async actions? I saw someone mention QStash but not sure how that plugs in to the event flow since you would still need to make a pg_net request to your redis backend based on the event...right?

**Comment by devdras** [comment]
> I'm unfortunately dealing with this as well. For me it's when there are 100 concurrent events (like bulk deleting 100 rows). I'm curious what alternatives you all are using for listening to postgres events and performing async actions? I saw someone mention QStash but not sure how that plugs in to the event flow since you would still need to make a pg_net request to your redis backend based on the event...right?
> 
> EDIT:
> 
> Has anyone implemented this as an alternative?
> 
> https://supabase.com/docs/guides/realtime/subscribing-to-database-changes
> 
> Theoretically I could just subscribe to the same tables in my backend and therefore only need to use 1 (or however many servers I deploy) persistent realtime connection for the same features? I suppose the issue would be concurrency since my two backend servers would receive the same event notification....

I just used a queue. I've used Upstash, BullMQ and Inngest. All are good. Whenever you update/insert the piece of data that you wanted to trigger, send an event to your queue. Your queue then does the thing you want to do. Basically my solution was to not use Supabase Webhooks. Imo they're useless without some kind of queue functionality backing them up.

**Comment by uncvrd** [comment]
@devdras Thanks for the quick response! I guess the issue that I'm having with that approach is that I need to listen for events from records that are deleted based on cascade events (say you have deleted an entity that has a fkey to a different table with onDelete CASCADE). Having to keep track of when a secondary entity is also removed when removing the original entity quickly becomes really messy.

That's why listening to the database row level events has been super powerful. A single source of truth that sends updates when a row is removed (whether directly or indirectly)

My guess is that I would need to use a KV store like Upstash or NATS to create a "lock" on an entity (based on the table and id) where you can only write to the queue if that key does not exist in the KV store. Maybe with having the key TTL being a couple minutes to remove from memory

Maybe that's what you're doing and now I'm understanding or not haha

Sorry to slightly derail the goal of this ticket but I hope it helps for people arriving at the same issue currently

**Comment by devdras** [comment]
@uncvrd I'd probably use a postgres function for that 🙂

**Comment by vickkhera** [comment]
> I just used a queue. I've used Upstash, BullMQ and Inngest. All are good. Whenever you update/insert the piece of data that you wanted to trigger, send an event to your queue. Your queue then does the thing you want to do. Basically my solution was to not use Supabase Webhooks. Imo they're useless without some kind of queue functionality backing them up.

My problem comes from reprocessing my internal queue which feeds Inngest... If I notice a request did not get accepted by Inngest, I retry it, but this is done with a single SQL statement more or less `SELECT resubmit_to_innngest(data) from queue_table where ...`.  When there are too many pg_net seems to just drop them so I have to retry yet again!

At some point pg_net has to be more reliable so I don't have to break this up into submitting no more than some small number at a time.

**Comment by GaryAustin1** [comment]
I was just testing using pg_net to realtime broadcast endpoint.
A bulk insert on a table of 20 rows works.  40 rows does not with timeout for all.
Simple SQL:
```SQL
insert into cats (count)
select * from generate_series (2300,2350)
```
The trigger function call the realtime broadcast end point.

There is a 2nd issue in that it appears pg_net does not guarantee order.  I'm assuming that it sends the http requests async (versus it is just async back to the SQL caller).  Because of this rapidly sent http requests can arrive out of order at their destination.

**Comment by JosephTico** [comment]
Issue still happening

**Comment by steve-chavez** [comment]
Can anyone confirm that this still happens on pg_net `0.10.0`? To confirm the version do:

```sql

select extversion from pg_extension where extname = 'pg_net';
```
| extversion |
| ---------- |
| 0.10.0     |

I'm no longer able to reproduce this previous error (https://github.com/supabase/pg_net/issues/86#issuecomment-1886040988) on Supabase cloud. We recently upgraded our libcurl version, which likely cleared the bug.

**Comment by vickkhera** [comment]
> Can anyone confirm that this still happens on pg_net `0.10.0`? To confirm the version do:

I want to test this out locally, but I cannot get that version in the CLI. I upgraded my preview environment last night, linked locally, and restarted so I do have the latest available docker images.

```
[Vicks-M1-Air]% npx supabase services


        SERVICE IMAGE      │      LOCAL       │   LINKED
  ─────────────────────────┼──────────────────┼─────────────
    supabase/postgres      │ 15.6.1.118       │ 15.6.1.118
    supabase/gotrue        │ v2.160.0         │ v2.160.0
    postgrest/postgrest    │ v12.2.1          │ -
    supabase/realtime      │ v2.30.34         │ -
    supabase/storage-api   │ v1.11.0          │ v1.11.0
    supabase/edge-runtime  │ v1.58.2          │ -
    supabase/studio        │ 20240729-ce42139 │ -
    supabase/postgres-meta │ v0.83.2          │ -
    supabase/logflare      │ 1.4.0            │ -
    supabase/supavisor     │ 1.1.56           │ -

[Vicks-M1-Air]% psql $PG_DEV
Timing is on.
Expanded display is used automatically.
psql (15.8 (Homebrew), server 15.6)
Type "help" for help.

postgres=> select extversion from pg_extension where extname = 'pg_net';
 extversion
------------
 0.8.0
(1 row)

Time: 4.371 ms
postgres=>
\q
[Vicks-M1-Air]% psql $PG_PREVIEW
Timing is on.
Expanded display is used automatically.
psql (15.8 (Homebrew), server 15.6)
SSL connection (protocol: TLSv1.3, cipher: TLS_AES_256_GCM_SHA384, compression: off)
Type "help" for help.

postgres=> select extversion from pg_extension where extname = 'pg_net';
 extversion
------------
 0.10.0
(1 row)

Time: 19.335 ms
```

Isn't the docker image supposed to bring the newer version?

**Comment by GaryAustin1** [comment]
@steve-chavez 
Limited testing.
With .10 and a 5 second timeout I can do a bulk insert of 1000 rows and send 1000 realtime broadcast endpoint calls from a trigger function and pg_net with no error. (tried twice). With 2 second timeout it fails in bunches of 100 or so working then not working.  With a 1 second time a batch of 100 failed at 30 (1 try and was first try).

I did not test at 5 seconds before so don't know if it is working better or not.

**Comment by nathanaeng** [comment]
Also experiencing this error for a batch size of 100.

**Comment by RTLS** [comment]
I did some testing in this area and found dropped requests due to ephemeral port exhaustion through docker networking. Limited testing of `pg_net` on bare metal did not reproduce dropped requests.

https://blog.sequinstream.com/benchmarking-pg_net-part-1/

**Comment by devdras** [comment]
> I did some testing in this area and found dropped requests due to ephemeral port exhaustion through docker networking. Limited testing of `pg_net` on bare metal did not reproduce dropped requests.
> 
> https://blog.sequinstream.com/benchmarking-pg_net-part-1/

Fantastic write up!

**Comment by khera** [comment]
> Isn't the docker image supposed to bring the newer version?

Looks like I had to do a `supabase db reset` to pick up the newer plugins on local dev.

**Comment by steve-chavez** [comment]
Solved in https://github.com/supabase/pg_net/pull/139.

**Comment by Riyer01** [comment]
I'm still facing this issue on hosted supabase -- any batch more than ~10 records will not trigger any webhooks.

**Comment by fan123199** [comment]
Yes, I am facing this issue also. Updated to version 0.10.0. Once updated, I think it resolved the performance issue. But when next peak time , webhooks still delays . (I think peak not very high , about 500/s insert) . And I found pg_net release version 0.11.0, how could I use it in supabase production right now?

**Comment by TheOtherBrian1** [comment]
@fan123199 pg_net v0.11.0 is not yet available on the hosted Supabase platform. It was merged into Supabase Postgres, but the new image still needs to be reviewed by the Infra Team before it will be widely released

**Comment by TheOtherBrian1** [comment]
@fan123199, what type of batch_size do you need? If possible, you could lower it to reduce strain on the background worker

**Comment by fan123199** [comment]
I want bigger batch_size. From docs, I think the bigger size, the more requests can be called per period.  Many row on net.http_request_queue waiting to be called, and more and more rows, and it delays. A trigger to call edge function action will be excute after 20minutes or more than one hour, it is not accepted. And I am in 12XL in supabase plan.

**Comment by ahanusek** [comment]
@fan123199 Check my hidden [comment](https://github.com/supabase/pg_net/issues/86#issuecomment-2014432863) in this thread. They don't care.

**Comment by fan123199** [comment]
> @fan123199 Check my hidden [comment](https://github.com/supabase/pg_net/issues/86#issuecomment-2014432863) in this thread. They don't care.

Thanks, after day and night, we now use planB to resolve the problem. We just made a similar pg_net implementation. 
1. Change webhook trigger to insert a request data to a new table queue_temp, similar column as net.http_request_queue
1. Buy a VPS  near our database
2. run a server (write in python) on VPS to read from queue_temp. (about 200 oldest rows a seconds), and then send request. and then delete from queue_temp.

Now It works well. 

We are still awaiting the official release of the latest Supabase version to address this issue. This critical bug affects small to medium-sized businesses and undermines our confidence in Supabase's performance as our business grows.

**Comment by steve-chavez** [comment]
This is now available on Supabase cloud :rocket:. 

Make sure to check your pg_net version:

```sql
select * from pg_available_extensions where name ilike '%pg_net%';
```

| name   | default_version | installed_version | comment    |
| ------ | --------------- | ----------------- | ---------- |
| pg_net | 0.13.0          | 0.13.0            | Async HTTP |
